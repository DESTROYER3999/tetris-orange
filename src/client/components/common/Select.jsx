import React from "react";
import Select from "@material-ui/core/Select";
import FormHelperText from "@material-ui/core/FormHelperText";
import FormControl from "@material-ui/core/FormControl";
import OutlinedInput from "@material-ui/core/OutlinedInput";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(theme => ({
  formControl: {
    minWidth: 150,
    marginRight: "10px",
    color: "white",
    "& .MuiOutlinedInput-root": {
      "& fieldset": {
        borderColor: "white"
      },
      "&:hover fieldset": {
        borderColor: "white"
      },
      "&.Mui-focused fieldset": {
        borderColor: "white"
      },
      "&.Mui-disabled fieldset": {
        borderColor: "white"
      }
    }
  },
  label: {
    padding: "0.5vh",
    fontSize: "2vh",
    color: "white"
  },
  select: {
    color: "white",
    "& .MuiSelect-icon": {
      color: "white"
    },
    "& .MuiSelect-root": {
      fontFamily: "ArcadeClassic"
    }
  }
}));

const SelectMenu = props => {
  const classes = useStyles();

  return (
    <FormControl className={classes.formControl}>
      <label className={classes.label}>{props.label}</label>
      <Select className={classes.select} {...props} input={<OutlinedInput />}>
        {props.children}
      </Select>
      {props.helperText ? (
        <FormHelperText style={{ height: "2vh" }}>
          {props.helperText}
        </FormHelperText>
      ) : null}
    </FormControl>
  );
};

export default SelectMenu;
