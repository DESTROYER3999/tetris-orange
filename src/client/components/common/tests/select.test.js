import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Select from "../Select";

describe("<Select />", () => {
  test("Select rendering without helperText", () => {
    const wrapper = shallow(<Select />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("Select rendering with helperText", () => {
    const wrapper = shallow(<Select helperText={"helperText"} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
