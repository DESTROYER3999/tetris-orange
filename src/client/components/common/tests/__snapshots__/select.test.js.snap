// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> Select rendering with helperText 1`] = `
<WithStyles(ForwardRef(FormControl))
  className="makeStyles-formControl-1"
>
  <label
    className="makeStyles-label-2"
  />
  <WithStyles(ForwardRef(Select))
    className="makeStyles-select-3"
    helperText="helperText"
    input={<ForwardRef(WithStyles) />}
  />
  <WithStyles(ForwardRef(FormHelperText))
    style={
      Object {
        "height": "2vh",
      }
    }
  >
    helperText
  </WithStyles(ForwardRef(FormHelperText))>
</WithStyles(ForwardRef(FormControl))>
`;

exports[`<Select /> Select rendering without helperText 1`] = `
<WithStyles(ForwardRef(FormControl))
  className="makeStyles-formControl-1"
>
  <label
    className="makeStyles-label-2"
  />
  <WithStyles(ForwardRef(Select))
    className="makeStyles-select-3"
    input={<ForwardRef(WithStyles) />}
  />
</WithStyles(ForwardRef(FormControl))>
`;
